### DISCUSSION TITLE
equal operation?

### DISCUSSION THREAD

**Discussion by yjqiang** [101arrowz/fflate#120] [General]
https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons
Maybe should change all `==` to `===` and `!=` to `!==`?

**Comment by 101arrowz** [comment]
I'm well aware of the differences between strict and loose equality operators, I chose the loose equality because I don't need the guarantees of strict equality and the loose operators save a byte each.

**Reply by yjqiang** [reply]
“ I don't need the guarantees of strict equality ”, I don't get it. Why? I guess because many functions(like [inflt](https://github.com/101arrowz/fflate/blob/master/src/index.ts#L238)) are defined by yourself and aren't exposed to the users(kind like private or protect), so the values' types can be controlled?

**Reply by 101arrowz** [reply]
Yes, exactly. I'm more careful around user-exposed functions (usually I'll only do `!= null` there, which just means not `undefined` and not `null`), but as I'm using TypeScript I don't really need to worry about accidentally comparing two invalid types within the private functions.
